plugins/sonicwall.yaml

```diff
@@ -0,0 +1,161 @@
+    default: true
+    advanced_config: true
+  - name: include_file_path
+    label: Include File Path
+    description: Include File Path as a label
+    type: bool
+    default: false
+    advanced_config: true
+  - name: include_file_name_resolved
+    label: Include Resolved File Name
+    description: Same as include_file_name, however, if file name is a symlink, the underlying file's name will be set as a label
+    type: bool
+    default: false
+    advanced_config: true
+  - name: include_file_path_resolved
+    label: Include Resolved File Path
+    description: Same as include_file_path, however, if file path is a symlink, the underlying file's path will be set as a label
+    type: bool
+    default: false
+    advanced_config: true
+# Set Defaults
+# {{$start_at := default "end" .start_at}}
+# {{$include_file_name := default true .include_file_name}}
+# {{$include_file_path := default false .include_file_path}}
+# {{$include_file_name_resolved := default false .include_file_name_resolved}}
+# {{$include_file_path_resolved := default false .include_file_path_resolved}}
+pipeline:
+  - id: log_reader
+    type: file_input
+    include:
+    # {{ range $i, $fp := .file_log_path  }}
+      - '{{ $fp }}'
+    # {{ end }}
+    exclude:
+    # {{ range $i, $efp := .exclude_file_log_path  }}
+        - '{{ $efp }}'
+    # {{ end }}
+    start_at: {{ $start_at }}
+    include_file_name: {{ $include_file_name }}
+    include_file_path: {{ $include_file_path }}
+    include_file_name_resolved: {{ $include_file_name_resolved }}
+    include_file_path_resolved: {{ $include_file_path_resolved }}
+    labels:
+      log_type: 'sonicwall'
+      plugin_id: {{ .id }}
+    write_to: log
+
+  - id: message_router
+    type: router
+    # Default to just move log entry to $record because it doesn't match known format
+    default: message_move 
+    routes:
+      # Check if log starts with a syslog priority.
+      # Example: <12>
+      - output: beginning_priority_parser
+        expr: '$record.log matches "^<[^>]+>\\s*"'
+      # Check if log starts with timestamp and ip address.
+      # Example Jan  3 13:45:50 192.168.5.1
+      - output: beginning_timestamp_parser
+        expr: '$record.log matches "^\\w{3}\\s+\\d{1,2}\\s+\\d{2}:\\d{2}:\\d{2}\\s+[:a-fA-F\\d\\.]+\\s+"'
+
+  # Parse priority, and message.
+  - id: beginning_priority_parser
+    if: '$record.log matches "^<[^>]+>\\s*[\\d\\D]*"'
+    type: regex_parser
+    parse_from: $record.log
+    regex: '^<(?P<priority>[^>]+)>\s*(?P<message>[\d\D]*)'
+    output: priority_severity_parser
+
+  - id: beginning_timestamp_parser
+    if: '$record.log matches "^\\w{3}\\s+\\d{1,2}\\s+\\d{2}:\\d{2}:\\d{2}\\s+"'
+    type: regex_parser
+    parse_from: $record.log
+    regex: '^(?P<timestamp>\w{3}\s+\d{1,2}\s+\d{2}:\d{2}:\d{2})\s+(?P<host>[:a-fA-F\d\.]+)\s+(?P<message>[\d\D]*)'
+    timestamp:
+      parse_from: $record.timestamp
+      layout_type: gotime
+      layout: 'Jan _2 15:04:05'
```

Comment:
We should support a timezone when timezones do not exist in the log format

Reply:
Done